lib/fizzy/parser.hpp

```diff
@@ -79,7 +79,9 @@ inline parser_result<std::vector<T>> parse_vec(const uint8_t* pos, const uint8_t
     std::tie(size, pos) = leb128u_decode<uint32_t>(pos, end);
 
     std::vector<T> result;
-    result.reserve(size);
+    // Do not reserve memory for `size` number of elements. It can be a lie
```

Comment:
How is this a lie?

Reply:
_Can be_. Literally `ffffffff04`.

Reply:
Well that's going to fail anyway, won't it? Can't we just impose a rational limit here?

Reply:
> Can be. Literally ffffffff04.

If we do merge this (as it shouldn't mean any real difference at this point), please update the comment to be more clear.

Reply:
It is going to report `parser_error` instead of crashing. A limit would require additional justification.

Reply:
>It is going to report parser_error instead of crashing. 

Yep, and that's what the spectests require probably (though they haven't failed -- so should report this upstream).

> A limit would require additional justification.

We have #8 for that, but I guess this specific case is not worth it.